pandas/core/tools/datetimes.py

```diff
@@ -403,6 +407,18 @@ def _convert_listlike_datetimes(
             arg = arg.tz_convert(None).tz_localize("utc")
         return arg
 
+    elif isinstance(arg_dtype, ArrowDtype) and arg_dtype.type is Timestamp:
+        # TODO: Combine with above if DTI/DTA supports Arrow timestamps
+        if utc:
+            # pyarrow uses UTC, not lowercase utc
+            if isinstance(arg, Index):
+                arg_array = cast(ArrowExtensionArray, arg.array)
+                arg = Index(arg_array._dt_tz_localize("UTC"))
+            else:
+                # ArrowExtensionArray
+                arg = arg._dt_tz_localize("UTC")
```

Comment:
is `tz_localize` always the right one, or do we need `tz_convert` if the input was timezone-aware to begin with?

In the test you've added, `dti_arrow` is always timezone-naive - would we make a timezone-aware example too please, just to check this?

Reply:
Thx for the catch.

I messed up the test (it does parametrize over UTC and US/Central), but I forgot to propagate the tz information in the tests.